apps/evm/src/clients/api/queries/getMerklRewards/formatMerklRewardsResponse.ts

```diff
@@ -0,0 +1,139 @@
+import BigNumber from 'bignumber.js';
+import type { MerklDistribution, Token } from 'types';
+import { areAddressesEqual } from 'utilities';
+import type { GetMerklUserRewardsResponse, MerklRewardBreakdown } from '.';
+import type { PendingExternalRewardSummary } from '../getPendingRewards/types';
+
+type MerklCampaign = MerklDistribution & { poolComptrollerAddress: string };
+
+export const formatMerklRewardsPayload = (
+  apiPayload: GetMerklUserRewardsResponse,
+  merklCampaignsPerPool: Record<string, Record<string, MerklDistribution[]>>,
+): PendingExternalRewardSummary[] => {
+  const allCampaigns = Object.keys(merklCampaignsPerPool).reduce<MerklCampaign[]>(
+    (acc, poolComptrollerAddress) => {
+      const merklDistributionsPerAsset = merklCampaignsPerPool[poolComptrollerAddress];
+      const assetDistributionArray = Object.keys(merklDistributionsPerAsset).reduce<
+        MerklCampaign[]
+      >(
+        (accPoolCampaigns, vTokenAddress) => [
+          ...accPoolCampaigns,
+          ...merklDistributionsPerAsset[vTokenAddress].map(d => ({
+            ...d,
+            poolComptrollerAddress,
+          })),
+        ],
+        [],
+      );
+      return [...acc, ...assetDistributionArray];
+    },
+    [],
+  );
+
+  // list all Merkl campaign ids and their corresponding reward token
+  const campaignIdRewardTokenMapping = allCampaigns.reduce<Record<string, Token>>(
+    (acc, d) => ({
+      ...acc,
+      [d.rewardDetails.merklCampaignIdentifier.toLowerCase()]: d.token,
+    }),
+    {},
+  );
+
+  // from the API response, filter the reward breakdowns from campaigns relevant to Venus
+  // it will only contain one entry at most, since we filtered by chainId
+  const apiRewards = apiPayload?.[0].rewards || [];
+
+  // apiRewards is an array of all the user reward tokens in the given chain
+  // these entries are generated from campaigns, and each campaign breakdown tells how much a user can claim/has claimed
+  const { flattenedBreakdowns } = apiRewards.reduce<{
+    flattenedBreakdowns: MerklRewardBreakdown[];
+  }>(
+    (acc, r) => ({
+      flattenedBreakdowns: [...acc.flattenedBreakdowns, ...r.breakdowns],
+    }),
+    {
+      flattenedBreakdowns: [],
+    },
+  );
+
+  const breakdownsFromVenusCampaigns = flattenedBreakdowns.filter(b =>
+    Object.keys(campaignIdRewardTokenMapping).includes(b.campaignId.toLowerCase()),
+  );
+
+  const campaignIdToBreakdownMapping = breakdownsFromVenusCampaigns.reduce<
+    Record<string, MerklRewardBreakdown | undefined>
+  >(
+    (acc, b) => ({
+      ...acc,
+      [b.campaignId.toLowerCase()]: b,
+    }),
+    {},
+  );
```

Comment:
You can simplify this logic:
```suggestion
  const campaignIdToBreakdownMapping: Record<string, MerklRewardBreakdown> = {};

  apiRewards.forEach(reward => {
    reward.breakdowns.forEach(breakdown => {
      const campaignId = breakdown.campaignId.toLowerCase();
      if (campaignIdRewardTokenMapping[campaignId]) {
        campaignIdToBreakdownMapping[campaignId] = breakdown;
      }
    });
  });
```